astropy/table/tests/test_table.py

```diff
@@ -3592,3 +3592,14 @@ def test_table_replace_column_with_scalar(empty_table):
     # Direct replacement should never work.
     with pytest.raises(ValueError, match="cannot replace.*with a scalar"):
         t.replace_column("a", 2)
+
+
+@pytest.mark.parametrize(
+    "arr", [None, [], [[], []], (), ((), ()), np.array([]), np.array([[], []])]
+)
+@pytest.mark.parametrize("arg_type", ["rows", "data"])
```

Comment:
@taldcroft , your patch fails with `data=np.array([[], []])` combo. Is that expected? If so, I can update the test. If not, then your suggested patch needs more... patching. Hope you can advise. Thanks!

Reply:
Yes, that is expected. This is an unfortunate duality in Table input which is discussed in the last bit of https://docs.astropy.org/en/stable/table/construct_table.html#numpy-homogeneous-array.

Basically `np.array()` is always interpreted as row-ordered, so `np.array([[], []])` is a 2-row table of 0 columns, which ends up being an empty table (zero rows and zero columns). No doubt this is confusing, but I think that special-casing this would be even worse.

Reply:
The upshot being that you need to exclude `np.array([[],[]])` from the `data` inputs.

Reply:
Thanks for the clarification! I updated the test. 🤞